src/libs/ReportUtils.ts

```diff
@@ -1813,10 +1813,9 @@ function canDeleteReportAction(reportAction: OnyxInputOrEntry<ReportAction>, rep
             return false;
         }
 
-        const linkedReport = isThreadFirstChat(reportAction, reportID) ? getReportOrDraftReport(report?.parentReportID) : report;
         if (isActionOwner) {
-            if (!isEmptyObject(linkedReport) && (isMoneyRequestReport(linkedReport) || isInvoiceReport(linkedReport))) {
-                return canDeleteTransaction(linkedReport);
+            if (!isEmptyObject(report) && (isMoneyRequestReport(report) || isInvoiceReport(report))) {
+                return canDeleteTransaction(report);
             }
```

Comment:
`canDeleteReportAction` is only being used in ContextMenuActions.

> const linkedReport = isThreadFirstChat(reportAction, reportID) ? getReportOrDraftReport(report?.parentReportID) : report;

This logic was added in https://github.com/Expensify/App/pull/40287 because the reportID of the thread parent message is the currently view report. But because the reportID now contains the reportID of the report action, we don't need this anymore.